### PR TITLE
committing changes for historian

### PR DESCRIPTION
**Pull request by SwapneelM**
The commit adds changes to the Eventbuffer, Subscription and Pipeline Models as well as changes test cases which restrict the creation of empty Eventbuffers. Support for Historian has been added to the Eventbuffer.

### CODE REVIEW

File: src/main/java/com/falkonry/client/service/FalkonryService.java [hunk 1]
```diff
@@ -24,30 +27,22 @@ public FalkonryService (String host, String token) throws Exception {
...
+            .setThingIdentifier(eventbuffer.getThingIdentifier())
+            .setTimeFormat(eventbuffer.getTimeFormat())
+            .setTimeIdentifier(eventbuffer.getTimeIdentifier())
+            .setSignalsTagField(eventbuffer.getSignalsTagField())
+            .setSignalsDelimiter(eventbuffer.getSignalsDelimiter())
+            .setValueColumn(eventbuffer.getValueColumn())
+            .setSignalsLocation(eventbuffer.getSignalsLocation());
+
+    String eventbuffer_json = httpService.post("/eventbuffer", mapper.writeValueAsString(eb));
+    System.out.println(eventbuffer_json);
```
Comment (aniket-amrutkar) [inline src/main/java/com/falkonry/client/service/FalkonryService.java]: remove sysout

File: src/main/java/com/falkonry/helper/models/Pipeline.java [hunk 2]
```diff
@@ -111,25 +111,30 @@ public Pipeline setInputMeasurement(String inputMeasurement) {
     return this;
   }
 
+
   public String getThingIdentifier() {
     return thingIdentifier;
   }
 
+  /*
```
Comment (aniket-amrutkar) [inline src/main/java/com/falkonry/helper/models/Pipeline.java]: remove commented code

File: src/test/java/com/falkonry/TestAddData.java [hunk 3]
```diff
@@ -15,52 +14,55 @@
 
 public class TestAddData {
     Falkonry falkonry = null;
-    String host = "http://localhost:8080";
-    String token = "";
+    String host = "https://dev.falkonry.io";
```
Comment (aniket-amrutkar) [inline src/test/java/com/falkonry/TestAddData.java]: replace this with back to localhost and remove authToken

File: src/test/java/com/falkonry/TestAddVerification.java [hunk 4]
```diff
@@ -14,35 +14,34 @@
 
 public class TestAddVerification {
     Falkonry falkonry = null;
-    String host = "http://localhost:8080";
-    String token = "";
+    String host = "https://dev.falkonry.io";
```
Comment (aniket-amrutkar) [inline src/test/java/com/falkonry/TestAddVerification.java]: Same here

File: res/output.txt [hunk 5]
```diff
@@ -0,0 +1,51 @@
+time,thing,value
```
Comment (aniket-amrutkar) [inline res/output.txt]: Why do we need this file in repo ?

File: res/verificationData_historian.csv [hunk 6]
```diff
@@ -0,0 +1,3 @@
+time,thing,end,Health
```
Comment (aniket-amrutkar) [inline res/verificationData_historian.csv]: change the file names.. these are not historian files.
this should be verificationData.csv

File: res/verificationData_historian.json [hunk 7]
```diff
@@ -0,0 +1,2 @@
+{"time" : "2011-03-26T12:00:00Z", "thing" : "thing1", "end" : "2012-06-01T00:00:00Z", "Health" : "Normal"}
```
Comment (aniket-amrutkar) [inline res/verificationData_historian.json]: this should be verificationData.json

File: src/main/java/com/falkonry/helper/models/Subscription.java [hunk 8]
```diff
@@ -145,10 +116,18 @@ public Boolean getHistorian() {
...
+    return archived;
+  }
+
   @JsonProperty("isHistorian")
   public Subscription setHistorian(Boolean historian) {
     this.isHistorian = historian;
     return this;
   }
 
+  public Subscription setArchived(Boolean archived) {
```
Comment (aniket-amrutkar) [inline src/main/java/com/falkonry/helper/models/Subscription.java]: We don't need setter of this variable.
Infact do we need getter also ?

Comment (SwapneelM) [inline src/main/java/com/falkonry/helper/models/Subscription.java]: Hi,

Thanks for the updates. I'll edit it as required and reopen a pull request
by EOD.

Regards,
Swapneel Mehta

File: src/main/java/com/falkonry/helper/models/Subscription.java [hunk 9]
```diff
@@ -23,6 +21,7 @@
   private String valueColumn;
   private Boolean streaming;
   private Boolean isHistorian;
+  private Boolean archived;
```
Comment (aniket-amrutkar) [inline src/main/java/com/falkonry/helper/models/Subscription.java]: remove isHistorian and archived.

**Pull request merged**